plugins/tileacq.py

```diff
@@ -165,49 +180,105 @@ def _update_total_area(self, _=None):
+        fovs = [self._get_fov(s) for s in ss]
         if not fovs:
             raise ValueError("No stream so no FoV, so no minimum one")
 
         return (min(f[0] for f in fovs),
                 min(f[1] for f in fovs))
 
     def show_dlg(self):
-        # TODO: only accept to run if in "secom_live" or "sparc_acqui" tab
-        # (iow, has a streambar_controller)
-        # TODO: support SPARC by using "sparc_acqui"
         # TODO: if there is a chamber, only allow if there is vacuum
 
-        tab = self.main_app.main_data.getTabByName("secom_live")
-        ss = tab.tab_data_model.streams.value
+        # Fail if the live tab is not selected
+        tab = self.main_app.main_data.tab.value
+        if tab.name not in ("secom_live", "sparc_acqui"):
+            box = wx.MessageDialog(self.main_app.main_frame,
+                       "Tiled acquisition must be done from the acquisition stream.",
+                       "Tiled acquisition not possible", wx.OK | wx.ICON_STOP)
+            box.ShowModal()
+            box.Destroy()
+            return
+
+        tab = self.main_app.main_data.tab.value
+
+        # If no roi is selected, select entire area
+        try:
+            if tab.tab_data_model.semStream.roi.value == UNDEFINED_ROI:
+                tab.tab_data_model.semStream.roi.value = (0, 0, 1, 1)
+        except AttributeError:
+            pass  # Not a SPARC
+
+        # Disable drift correction
+        if hasattr(tab.tab_data_model, "semStream"):
+            # SPARC
+            tab.tab_data_model.semStream.dcRegion.value = UNDEFINED_ROI
 
+        ss = self._get_live_streams(tab.tab_data_model)
         self.filename.value = self._get_new_filename()
 
         dlg = AcquisitionDialog(self, "Tiled acquisition",
                                 "Acquire a large area by acquiring the streams multiple "
                                 "times over a grid.")
+
         self._dlg = dlg
         dlg.addSettings(self, self.vaconf)
         for s in ss:
             dlg.addStream(s)
         dlg.addButton("Cancel")
         dlg.addButton("Acquire", self.acquire, face_colour='blue')
 
-        # Update acq time and area when streams are added/removed
+        # Update acq time and area when streams are added/removed. Add stream settings
+        # to subscribed vas.
         dlg.microscope_view.stream_tree.flat.subscribe(self._update_exp_dur, init=True)
         dlg.microscope_view.stream_tree.flat.subscribe(self._update_total_area, init=True)
-        # TODO: also update when settings change
+        dlg.microscope_view.stream_tree.flat.subscribe(self._on_streams_change, init=True)
+
+        self.nx.subscribe(self._check_range, init=True)
+        self.ny.subscribe(self._check_range, init=True)
 
         # TODO: disable "acquire" button if no stream selected.
 
         ans = dlg.ShowModal()
-        if ans == 0:
+        if ans == 0 or ans == 5101:  # 5101 code returned after exiting with top left close button
```

Comment:
Don't hardcode `5101`, that's an event ID from wx. Probably something like `wx.ID_CLOSE`.

Reply:
In addition, you might want to check if the acquisition is cancelled in such case. If not, (and the acquisition is actually running) you'd need to cancel it now, before closing the window.